src/expr.rs

```diff
@@ -1909,6 +1909,17 @@ pub mod parsing {
         }
     }
 
+    #[cfg(feature = "full")]
+    impl Parse for ExprRange {
+        fn parse(input: ParseStream) -> Result<Self> {
+            let expr: Expr = input.parse()?;
+            match expr {
+                Expr::Range(expr_range) => Ok(expr_range),
+                _ => Err(input.error("expected range expression"))
```

Comment:
This would incorrectly point to the token after the expression that just got parsed, since that is where `input` would be. It would be better as:

```rust
Err(Error::new_spanned(expr, "..."))
```